src/algebra/group_power/lemmas.lean

```diff
@@ -277,6 +284,16 @@ begin
   exact lt_irrefl _ (lt_of_le_of_lt (nsmul_le_nsmul (le_of_lt ha) $ not_lt.mp H) h)
 end
 
+lemma eq_of_gsmul_eq_gsmul_of_pos {a b : A} {m : ℤ} (hab : m • a = m • b) (hm : 0 < m) : a = b :=
```

Comment:
I think you should add the lemma
```lean
lemma gsmul_eq_gsmul_iff {a b : A} {m : ℤ} (hm : 0 < m) : m • a = m • b ↔ a = b :=
```
Where the forward direction is this proof, and the reverse direction is trivial.

Reply:
Sounds good!